addons/event/models/event_event.py

```diff
@@ -494,13 +505,19 @@ def _compute_ticket_instructions(self):
                is_html_empty(event.event_type_id.ticket_instructions):
                 event.ticket_instructions = event.event_type_id.ticket_instructions
 
-    @api.constrains('seats_max', 'seats_available', 'seats_limited')
-    def _check_seats_limit(self):
+    @api.constrains('seats_max', 'seats_limited', 'registration_ids')
+    def _check_seats_availability(self, n_registrations=0):
+        """Used either
+        * as classic constraint
+        * to test before trying to confirm one or several registrations that
+           there are enough seats available for this event."""
         for event in self:
-            if event.seats_limited and event.seats_max and event.seats_available < 0:
-                raise ValidationError(_('No more available seats for %s. '
-                                        'Raise the limit or remove some other confirmed registrations first.',
-                                        event.name))
+            if (event.seats_limited and event.seats_max
+                    and event.seats_available < n_registrations):
+                raise ValidationError(
+                    _('There are no more available seats for %s. '
```

Comment:
``There are no more available seats`` or ``There is not any seat available anymore`` ? Like they say, pardon my english, asking to be sure we use correct wording ^^